### ISSUE TITLE
No interaction on websites with frames

### ISSUE THREAD

**Issue by pjoubert-** [cmcaine/tridactyl#122]
Besides frames are bad, some websites still use them. I use some of them, where keyboard interactions are just not working at all.

Example website: http://www.angelfire.com/super/badwebs/

tridactyl version: 1.5.1pre409

**Comment by Bzly** [comment]
This is already noted directly (albeit titled less accurately) in #67 and indirectly in #28. 

From the former:

>This is an iframe issue: tridactyl currently loads into only the top level frame.
>
>We're aware of this issue and expect to fix it Soon TM. :)
> @cmcaine

**Comment by lisael** [comment]
Hi, and thank you for the amazing work.

Besides the fact that only current iframe links are shown with 'f', the main drawback is that we can't navigate between iframes. My trackpad is somehow broken. I sometimes get stuck in a menu or ToC frame and I can't read the content of the website without plugging a mouse. The first quick-win, to me is "just" iframes navigation, relaoding tridactyl in each visited iframe (I have no idea of the amount of work but it looks easier than sharing data among iframes)

**Comment by cmcaine** [comment]
Thanks for commenting :)

Can you provide us with a link to the site that features the behaviour you describe?

Unfortunately I don't think the quick-win you describe will work: hints wouldn't be emitted properly on multi-frame sites and other issues.

The way to fix it is to bite the bullet and teach Tridactyl about frames properly, but that requires a bunch of changes in messaging and elsewhere to make settings and logic tab or page specific (probably page specific).

@Koushien, your excmd changes would be useful here, actually.

**Comment by bovine3dom** [comment]
Unfortunately this was not fixed in #320.

**Comment by bovine3dom** [comment]
Scrolling up on http://docs.nvidia.com/cuda/parallel-thread-execution/index.html now scrolls up the table of contents, the page, and then the table of contents again, which is less than ideal.

**Comment by glacambre** [comment]
Yup, I'm aware of this issue but I think https://github.com/cmcaine/tridactyl/issues/35 is more appropriate for this.

However you're right that we still have issues with frames. https://github.com/cmcaine/tridactyl/issues/412 is an example and maybe https://github.com/cmcaine/tridactyl/issues/417 too? But since they're more specific about the problem (keyevents) and since scrolling/following hints on the website mentioned by @pjoubert- works quite well now I thought we could close this issue.

**Comment by FrankEular** [comment]
Will we implement the focus back/forward like `[f` and `]f` in vimperator?

**Comment by glacambre** [comment]
@FrankEular Are you talking about `[[` and `]]` which are bound to `:followpage prev` and `:followpage next` by default?

**Comment by FrankEular** [comment]
> @FrankEular Are you talking about `[[` and `]]` which are bound to `:followpage prev` and `:followpage next` by default?

No, I mean switching the focus of frames. Take the link of the original post as an example: http://www.angelfire.com/super/badwebs/

There are several frames. As far as I remember, I could take advantage of `[f` and `]f` in vimperator to switch between different frames. The upshot of this switching is that, say, I can scroll down/up in the frame.
